File: python/9th/test_2.py
'''
Author       : Daniel_Elendeer
Date         : 2020-11-21 18:33:18
LastEditors  : Daniel_Elendeer
LastEditTime : 2020-11-21 18:55:46
Description  :
'''

if __name__ == '__main__':
    n, k = map(int, input().split())
    lst = list(range(1, n + 1))
    print(lst)

    idx = 0
    count = k
    while len(lst) != 1:
        count -= 1

        if idx >= len(lst):
            idx = 0

        if count == 0:
            lst.remove(lst[idx])
            print(lst)
            count = k
        else :
            idx += 1

    print(lst.pop())

File: dataStructure/CAnalyse/test/TEST_2.cpp
/*********************************************
 * @Author       : Daniel_Elendeer
 * @Date         : 2020-05-07 16:38:46
 * @LastEditors  : Daniel_Elendeer
 * @LastEditTime : 2020-07-21 23:55:09
 * @Description  :
*********************************************/
#include <iostream>
#include <cstdio>

using namespace std;

const int N = 20; // 1e5 + 10;
const int M = 1e6 + 10;

int ne[N];
char s[M], p[N];

int main() {
    int n, m;
    cin >> n >> p + 1 >> m >> s + 1;

    ne[1] = 1;
    for (int i = 2, j = 1; i <= n; ++ i ) {
        while ((j - 1) && p[j] != p[i]) j = ne[j];
        if (p[j] == p[i]) j ++ ;
        ne[i] = j;
    }

    for (int i = 1, j = 1; i <= m; ++ i ) {
        while (j - 1 && p[j] != s[i]) j = ne[j];
        if (s[i] == p[j]) j ++ ;
        if (j - 1 == n) {
            printf("%d ", i - n);
            j = ne[j];
        }
    }
    return 0;
}